File: java/visitor/NightShiftScheduleManagement.java
package visitor;

public class NightShiftScheduleManagement implements ScheduleManagement {
    @Override
    public void generateReport() {
        System.out.println("Generating report for night shift...");
    }

    @Override
    public void calculateOverTime() {
        System.out.println("Calculating over time for night shift..");
    }

    @Override
    public void accept(ScheduleManagementVisitor scheduleManagementVisitor) {
        scheduleManagementVisitor.visit(this);
    }

}

File: C#/Visitor/NightShiftScheduleManagement.cs
namespace Visitor;

public sealed class NightShiftScheduleManagement : IScheduleManagement
{
    public void GenerateReport()
        => Console.WriteLine("Generating report for night shift...");

    public void CalculateOverTime()
        => Console.WriteLine("Calculating over time for night shift..");

    public void Accept(IScheduleManagementVisitor scheduleManagementVisitor)
        => scheduleManagementVisitor.Visit(this);
}
